### ISSUE TITLE
Warning: Accessing non-existent property 'find' of module exports inside circular dependency

### ISSUE THREAD

**Issue by LukasAV** [mapbox/node-pre-gyp#584]
Hi,

with v1.0.4 I am getting the following warning (when using the `--trace-warnings` flag):

```
(node:15343) Warning: Accessing non-existent property 'find' of module exports inside circular dependency
    at emitCircularRequireWarning (internal/modules/cjs/loader.js:650:11)
    at Object.get (internal/modules/cjs/loader.js:664:5)
    at Object.<anonymous> (/project-path/node_modules/@mapbox/node-pre-gyp/lib/node-pre-gyp.js:53:40)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at Module.require (/project-path/node_modules/require-in-the-middle/index.js:43:24)
    at require (internal/modules/cjs/helpers.js:88:18)
```

Any suggestions how to resolve this warning?

**Comment by bmacnaughton** [comment]
@lukasAV - thank you for the report and stack trace.

The problem is that node 14 warns about circular dependencies; i believe that this is related to the esm module work.

It can be fixed by replacing:

`exports.find = require('./pre-binding').find;`

with:

```
Object.defineProperty(exports, 'find', {
  get: function() {
    return require('./pre-binding').find;
  },
  enumerable: true
});
```

so that `pre-binding` is not required until after the module is loaded. but i want to spend more time on it to see if the circular dependency can be removed completely. i'll be able to get to that this weekend. In the meantime it's just a warning and, while annoying, doesn't prevent loading the file.

@springmeyer  - feel free to make the change above if you believe that this cannot wait until the weekend.

**Comment by springmeyer** [comment]
Thanks @bmacnaughton - seems fine to wait till weekend as this is only a warning.

**Comment by LukasAV** [comment]
Thanks for the quick reply and the provided fix. Great to know it will be fixed in the near future.

**Comment by bmacnaughton** [comment]
released in v1.0.5
